util/add-depends.pl

```diff
@@ -6,46 +6,248 @@
 # in the file LICENSE in the source distribution or at
 # https://www.openssl.org/source/license.html
 
+use strict;
+use warnings;
+
 use lib '.';
 use configdata;
 
+use File::Spec::Functions qw(:DEFAULT rel2abs);
 use File::Compare qw(compare_text);
 
+my $debug = $ENV{ADD_DEPEND_DEBUG};
 my $buildfile = $config{build_file};
-my $buildfile_new = "$buildfile-$$";
+my $build_mtime = (stat($buildfile))[9];
+my $rebuild = 0;
 my $depext = $target{dep_extension} || ".d";
-my @deps =
+my @depfiles =
     sort
-    grep { -f $_ }
+    grep {
+        # This grep has side effects.  Not only does if check the existence
+        # of the dependency file given in $_, but it also checks if it's
+        # newer than the build file, and if it is, sets $rebuild.
+        ${^WIN32_SLOPPY_STAT} = 1; # performance that doesn't affect mtime
```

Comment:
Why not in outer scope? I.e. once for all, as opposite to each iteration.

Reply:
You mean, other than my sloppy thinking (seem fitting, eh? ;-))?  No reason whatsoever.

Will fix.